### PR TITLE
diode plotting notebook examples

### PR DESCRIPTION
**Pull request by simbilod**
Examples on how to plot mesh data in a Jupyter notebook, with package installation instructions

GIF

### CODE REVIEW

**Comment by tcaduser** [comment]
Hi @simbilod , sorry I haven't looked at this yet.  I think your work is interesting with this and @gdsfactory.  I suppose I could put this in some type of contributed script directory in the main distribution, or maybe at https://github.com/devsim/devsim_misc

could you please put it in:
examples/plotting

and put a copyright or license number there too?

Please let me know your thoughts.

**Comment by simbilod** [comment]
This looks like a great place to put it, I'll repackage my code and make a PR there

**Comment by tcaduser** [comment]
Hi @simbilod, can you put the PR on the r2.2.0 branch?

**Comment by simbilod** [comment]
Hi @tcaduser , this has dropped off my radar until I saw the notification from this being closed :) I updated the branch and moved the example as you prescribed

Also, I've found some time to rework my meshing-from-gds solution, it's starting to take good shape (see https://github.com/gdsfactory/gdsfactory/pull/826). Next step is to automate basic DEVSIM configurations on such meshes.

**Comment by tcaduser** [comment]
Hi @simbilod 

Thanks for the update.  I check out the new pull request soon.

I'm very interested in your meshing solution.  I have scripts that can be used to add interfaces and contacts to gmsh generated structures:
https://github.com/devsim/devsim_misc/tree/main/gmsh

I also have scripts to create a background mesh for use to refine the mesh:
https://github.com/devsim/devsim_misc/tree/main/refinement

I think that simulation guided mesh refinement is a good approach for improving quality of results.

**Comment by simbilod** [comment]
Thanks for the links! You put a lot of work into this, and (at least for me) the gmsh API was hard to navigate.

Early on I tried to use the gmsh OCC kernel as you do, but I ran into issues where BooleanFragments would destroy my physicals. There's a bunch of issues on the GMSH GitLab (some very recent) and official examples that are aware of this, and the recommended fix seemed to be to retag elements after the operation using a bounding box. But especially in our field of photonics shapes can be quite arbitrary, that was difficult.

Defining entities that only overlap at their interfaces and using Coherence improved this a bit. Since Coherence is BooleanFragments only applied to the highest dimensional objects, it does not destroy the highest-dim physicals. However, that does not apply to lower-dimensional entities, e.g. you lose the interfaces and have to recover them.

Since a GDS is a 2.5D structured description, I opted instead to use GMSH's traditional non-CAD approach. I use the GDS polygons and gdsfactory LayerStack to define points, and from pairs of points line segments, then lines, then curved loops, then surfaces, and (this is in a dev branch) shells and volumes. There's a bit of pre-processing at the input to break up the GDS mask and LayerStack into a unique set of non-overlapping polygons and lines, but after that's done as I build GMSH entities I keep track of the entities that have been placed and reuse them as needed. So at the end I have a good map of where everything is and what GDS layer it originally corresponded to.

TL;DR maybe I should have tried your code first before developing mine :P
